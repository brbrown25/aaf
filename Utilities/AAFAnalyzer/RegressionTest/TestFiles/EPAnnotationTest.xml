<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE aaf-file SYSTEM "EPDerivationChain.dtd">

<aaf-file>

    <!-- Test UserComments property -->
    <top-level name="User Comments in Mob Slot Segment">
        <timeline-mob-slot edit-rate-denominator="1" edit-rate-numerator="1">
            <source-clip track-type="none">
                <comment value="Should Pass" name="Test Case 1"/>
                <lower-level>
                    <timeline-mob-slot edit-rate-denominator="1" edit-rate-numerator="1">
                        <eoc track-type="none"/>
                    </timeline-mob-slot>
                </lower-level>
            </source-clip>
        </timeline-mob-slot>
    </top-level>
    
    <top-level name="User Comments in CommentMarker">
		<event-mob-slot edit-rate-denominator="1" edit-rate-numerator="1">
            <sequence track-type="none">
				<comment-marker track-type="none">
                    <comment value="Should Pass" name="Test Case 2"/>
                </comment-marker>
            </sequence>
        </event-mob-slot>
    </top-level>
    
    <top-level name="Illegal User Comments">
        <timeline-mob-slot edit-rate-denominator="1" edit-rate-numerator="1">
			<sequence track-type="none">
                <sequence track-type="none">
                    <comment value="Should Fail" name="Test Case 3"/>
                    <eoc track-type="none"/>
                </sequence>
            </sequence>
        </timeline-mob-slot>
    </top-level>

	<!-- Test CommentMarkes in essence/non-essence tracks -->
    <!-- Note: Case 2 above tests CommentMarkers in non-essence tracks -->
    <top-level name="CommentMarker in Essence Track">
		<event-mob-slot edit-rate-denominator="1" edit-rate-numerator="1">
            <sequence track-type="audio">
				<comment-marker track-type="audio"/>
            </sequence>
        </event-mob-slot>
    </top-level>
    
    <top-level name="DescriptiveMarker in Essence Track">
		<event-mob-slot edit-rate-denominator="1" edit-rate-numerator="1">
            <sequence track-type="audio">
				<descriptive-marker track-type="audio"/>
            </sequence>
        </event-mob-slot>
    </top-level>

	<!-- NEED TO TEST A DESCRIPTIVE MARKER IN AN ESSENCE TRACK -->
    
    <!-- Test CommentMarkes in Mob Slots (and nested in mob slots) -->
    <!-- Note: "CommentMarker in Essence Track" above test tests nested in an event mob slot -->
    <top-level name="CommentMarker in Event Mob Slot">
		<event-mob-slot edit-rate-denominator="1" edit-rate-numerator="1">
			<comment-marker track-type="none"/>
        </event-mob-slot>
    </top-level>
    
    <top-level name="CommentMarker nested in Timeline Mob Slot">
		<timeline-mob-slot edit-rate-denominator="1" edit-rate-numerator="1">
            <sequence track-type="none">
				<comment-marker track-type="none"/>
            </sequence>
        </timeline-mob-slot>
    </top-level>
    
    <top-level name="CommentMarker in Timeline Mob Slot">
		<timeline-mob-slot edit-rate-denominator="1" edit-rate-numerator="1">
			<comment-marker track-type="none"/>
        </timeline-mob-slot>
    </top-level>
    
	<top-level name="CommentMarker nested in Static Mob Slot">
		<static-mob-slot>
            <sequence track-type="none">
				<comment-marker track-type="none"/>
            </sequence>
        </static-mob-slot>
    </top-level>
    
    <top-level name="CommentMarker in Static Mob Slot">
		<static-mob-slot>
			<comment-marker track-type="none"/>
        </static-mob-slot>
    </top-level>
    
    <!-- Note: "DescriptiveMarker in Essence Track" above test tests nested in an event mob slot -->
    <top-level name="DescriptiveMarker in Event Mob Slot">
		<event-mob-slot edit-rate-denominator="1" edit-rate-numerator="1">
			<descriptive-marker track-type="none"/>
        </event-mob-slot>
    </top-level>
    
    <top-level name="DescriptiveMarker nested in Timeline Mob Slot">
		<timeline-mob-slot edit-rate-denominator="1" edit-rate-numerator="1">
            <sequence track-type="none">
				<descriptive-marker track-type="none"/>
            </sequence>
        </timeline-mob-slot>
    </top-level>
    
    <top-level name="DescriptiveMarker in Timeline Mob Slot">
		<timeline-mob-slot edit-rate-denominator="1" edit-rate-numerator="1">
			<descriptive-marker track-type="none"/>
        </timeline-mob-slot>
    </top-level>
    
	<top-level name="DescriptiveMarker nested in Static Mob Slot">
		<static-mob-slot>
            <sequence track-type="none">
				<descriptive-marker track-type="none"/>
            </sequence>
        </static-mob-slot>
    </top-level>
    
    <top-level name="DescriptiveMarker in Static Mob Slot">
		<static-mob-slot>
			<descriptive-marker track-type="none"/>
        </static-mob-slot>
    </top-level>

</aaf-file>
