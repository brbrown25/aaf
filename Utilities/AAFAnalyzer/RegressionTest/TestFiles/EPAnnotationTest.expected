        Name:   Edit Protocol Annotation Test
        Desc:   Verify that all annotations in an AAF file are legal.
        Result: Failed
        Reason: 
        Passing Requirements: REQ_EP_151; REQ_EP_152
        Warning Requirements: None
        Failing Requirements: REQ_EP_147; REQ_EP_149; REQ_EP_150

            Name:   Use of Component::UserComments
            Desc:   Component::UserComments shall only be used on: the Segment of a MobSlot or a CommentMarker (or its subclasses).
            Result: Failed
            Reason: Mob slot with ID = 1 of Top-Level Composition "Illegal User Comments" contains a Component that illegally uses the Component::UserComments property.
            Passing Requirements: None
            Warning Requirements: None
            Failing Requirements: REQ_EP_147

            Name:   Use of CommentMarkers for Annotations
            Desc:   CommentMarkers may be used for annotations that are not associated with a specific essence track; otherwise DescriptiveMarkers (or its subclasses) shall be used.
            Result: Failed
            Reason: Mob slot with ID = 1 of Top-Level Composition "CommentMarker in Essence Track" is an essence track and uses CommentMarkers for annotations.
            Passing Requirements: None
            Warning Requirements: None
            Failing Requirements: REQ_EP_149

            Name:   Comment Marker Placement
            Desc:   CommentMarkers shall be placed only in EventMobSlots.  DescriptiveMarkers may be placed in any kind of MobSlot (e.g. TimelineMobSlot, EventMobSlot, StaticMobSlot).
            Result: Failed
            Reason: Mob slot with ID = 1 of Top-Level Composition "CommentMarker nested in Timeline Mob Slot" is not an Event Mob Slot but contains a CommentMarker.
            Passing Requirements: None
            Warning Requirements: None
            Failing Requirements: REQ_EP_150

            Name:   Comment Marker Placement
            Desc:   CommentMarkers shall be placed only in EventMobSlots.  DescriptiveMarkers may be placed in any kind of MobSlot (e.g. TimelineMobSlot, EventMobSlot, StaticMobSlot).
            Result: Failed
            Reason: Mob slot with ID = 1 of Top-Level Composition "CommentMarker in Timeline Mob Slot" is not an Event Mob Slot but contains a CommentMarker.
            Passing Requirements: None
            Warning Requirements: None
            Failing Requirements: REQ_EP_150

            Name:   Comment Marker Placement
            Desc:   CommentMarkers shall be placed only in EventMobSlots.  DescriptiveMarkers may be placed in any kind of MobSlot (e.g. TimelineMobSlot, EventMobSlot, StaticMobSlot).
            Result: Failed
            Reason: Mob slot with ID = 1 of Top-Level Composition "CommentMarker nested in Static Mob Slot" is not an Event Mob Slot but contains a CommentMarker.
            Passing Requirements: None
            Warning Requirements: None
            Failing Requirements: REQ_EP_150

            Name:   Comment Marker Placement
            Desc:   CommentMarkers shall be placed only in EventMobSlots.  DescriptiveMarkers may be placed in any kind of MobSlot (e.g. TimelineMobSlot, EventMobSlot, StaticMobSlot).
            Result: Failed
            Reason: Mob slot with ID = 1 of Top-Level Composition "CommentMarker in Static Mob Slot" is not an Event Mob Slot but contains a CommentMarker.
            Passing Requirements: None
            Warning Requirements: None
            Failing Requirements: REQ_EP_150

            Name:   Use of Component::UserComments
            Desc:   Component::UserComments shall only be used on: the Segment of a MobSlot or a CommentMarker (or its subclasses).
            Result: Failed
            Reason: Mob slot with ID = 1 of Top-Level Composition "Heredity Test 1" contains a Component that illegally uses the Component::UserComments property.
            Passing Requirements: None
            Warning Requirements: None
            Failing Requirements: REQ_EP_147

            Name:   Use of CommentMarkers for Annotations
            Desc:   CommentMarkers may be used for annotations that are not associated with a specific essence track; otherwise DescriptiveMarkers (or its subclasses) shall be used.
            Result: Failed
            Reason: Mob slot with ID = 1 of Unnamed Clip is an essence track and uses CommentMarkers for annotations.
            Passing Requirements: None
            Warning Requirements: None
            Failing Requirements: REQ_EP_149

